client/utils.ts

```diff
@@ -29,21 +29,19 @@ export const getNearbyFacilities = (
 
 		const relevantFacilities = facilities.filter((facility) => {
 			const { id, methodsAccepted, categoriesAccepted, latitude, longitude } = facility;
-			if (methodsAccepted.includes(method) && categoriesAccepted.includes(cat)) {
-				const distance = calculateDistance(
-					Number(address.coordinates.lat),
-					Number(address.coordinates.long),
-					latitude,
-					longitude,
-				);
-
-				if (distance < maxDistance) {
-					distances.set(id, distance);
-					allFacilityIds.push(id);
-					return true;
-				}
-			}
-			return false;
+			if (!methodsAccepted.includes(method) || !categoriesAccepted.includes(cat)) return false;
+			const distance = calculateDistance(
+				Number(address.coordinates.lat),
+				Number(address.coordinates.long),
+				latitude,
+				longitude,
+			);
+
+			if (distance >= maxDistance) return false;
```

Comment:
maxDistance is used to dynamically display facilities within x distance (alter maxDistance via Map Filter UI); for some reason, this code breaks the ability to change the maxDistance

![image](https://github.com/bettersg/recyclegowhere/assets/79783660/0d950604-d725-4d96-b5b1-b187310bfa9d)